### PR TITLE
Allow configuration of Base URL via website

### LINKED ISSUE #5246: [BUG] Instance update base_url
I migrated my dockerized FreshRSS instance, migration works fine but the base_url still points to the server where it was initially configured on

Migrate docker data to new server and run the instance on a new server

**Expected behavior**
base_url recognizing that it has been changed or a way to configure base_url from the webpage

### PR DESCRIPTION
**Pull request by samc1213**
Closes #5246 

Changes proposed in this pull request:

- Add a configuration for base_url in the System Configuration page

How to test the feature manually:

1. Open System Configuration
2. Change "Base URL"
3. Refresh page, and ensure Base URL is updated
4. Check config.php and ensure `base_url` is updated

Pull request checklist:

- [x] clear commit messages
- [x] code manually tested
- [ ] unit tests written (optional if too hard)
- [ ] documentation updated

[Additional information can be found in the documentation](https://github.com/FreshRSS/FreshRSS/tree/edge/docs/en/developers/04_Pull_requests.md).

### CODE REVIEW

File: app/i18n/fr/admin.php [hunk 1]
```diff
@@ -159,6 +159,7 @@
 	'system' => array(
 		'_' => 'Configuration du système',
 		'auto-update-url' => 'URL du service de mise à jour',
+		'base-url' => 'Base URL',	// TODO
```
Comment (Alkarex) [inline app/i18n/fr/admin.php]:
```suggestion
		'base-url' => 'URL de la racine',
```

File: app/views/configure/system.phtml [hunk 2]
```diff
@@ -54,6 +54,14 @@
 			</div>
 		</div>
 
+		<div class="form-group">
+			<label class="group-name" for="base-url"><?= _t('admin.system.base-url') ?></label>
+			<div class="group-controls">
+				<input type="text" id="base-url" name="base-url" value="<?= FreshRSS_Context::$system_conf->base_url ?>"
+					data-leave-validation="<?= FreshRSS_Context::$system_conf->limits['cookie_duration'] ?>"/>
```
Comment (Alkarex) [inline app/views/configure/system.phtml]:
```suggestion
					data-leave-validation="<?= FreshRSS_Context::$system_conf->base_url ?>"/>
```

**Comment by Alkarex** [comment]
Thanks, and welcome 👍🏻

File: app/views/configure/system.phtml (hunk 2 above)
Comment (samc1213) [reply app/views/configure/system.phtml]: Oops, good find, thank you

**Comment by Alkarex** [comment]
I have added an automatic detection hint:
image

**Review by Alkarex — APPROVED** [review]

**Review by Frenzie — APPROVED** [review]
makes sense

**Comment by samc1213** [comment]
> I have added an automatic detection hint: image

Love this idea - I tested it as well and it looks good to me.

**Comment by math-GH** [comment]
I am a bit late to the party...
I would expect an improved documentation and would like to have here a warning message. It is very easy to destroy the application.

I changed `http://localhost/freshrss/p` to `http://localhost/freshrss` and it crashed the system.

grafik

It is to easy to fail here

**Comment by Frenzie** [comment]
True, something like this might need a new UI element. Or at least I don't think we have something like an input disabled with an unlock button next to it anywhere.

I mean exactly the same thing as the show password button basically, just to get rid of the disabled attribute. You could additionally add a little warning underneath the title if that's not sufficiently saying "dangerous" all by itself.
image

**Comment by Alkarex** [comment]
Right, indeed @math-GH . This went a bit too fast indeed, and should have waited for 1.23.
Temporary mitigation https://github.com/FreshRSS/FreshRSS/pull/5657

**Comment by math-GH** [comment]
Let me think loud:
If the base URL is wrong, then the front end does not work. The admin cannot change it because (s)he cannot access.
If the base URL is correct, then it does not need to be changed.

What is the use case that needs the option? (I read the issue ticket but does not understand the use case)

**Comment by Frenzie** [comment]
> If the base URL is wrong, then the front end does not work. The admin cannot change it because (s)he cannot access.
If the base URL is correct, then it does not need to be changed.

This is incorrect. I had a wrong base URL for years before I even noticed, unless I'm thinking of a different similar setting.

**Comment by math-GH** [comment]
> > If the base URL is wrong, then the front end does not work. The admin cannot change it because (s)he cannot access.
> > If the base URL is correct, then it does not need to be changed.
> 
> This is incorrect. I had a wrong base URL for years before I even noticed, unless I'm thinking of a different similar setting.

That leads us back to the documentation: for what is the base URL used/needed?

**Comment by Alkarex** [comment]
The base URL is used for the generation of various addresses, including relative addresses, some redirections, and for constructing absolute URLs as needed by e.g. API, WebSub

**Comment by math-GH** [comment]
Any example for my setup that will work (default: `http://localhost/freshrss/p`)?

**Comment by Alkarex** [comment]
@math-GH An example of place where you can see it in action is in one of the RSS outputs. Check the `<atom:link href="".../>`

**Comment by Frenzie** [comment]
I believe in my case it wasn't wrong per se but the IP address instead of the domain which led to some subtle problem with something API-related iirc, like the inability to log on when entering the domain in a client.

**Comment by marien-probesys** [comment]
What about guessing the recommended `base_url` with JavaScript instead? `window.location.href` would return a reliable URL, just having to strip the query part.

**Comment by Alkarex** [comment]
> What about guessing the recommended `base_url` with JavaScript instead? `window.location.href` would return a reliable URL, just having to strip the query part.

Indeed Marien, as a tip to autofill the information in the configuration page. But that cannot be used the rest of the time (e.g. from API, WebSub, HTTP redirects) as the request is not passing through a JavaScript client environment.

**Pull request merged**